backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -154,21 +166,38 @@
     // Note: This method uses a stateless session
     public Uni<List<Endpoint>> getTargetEndpoints(String tenant, EventType eventType) {
         String query = "SELECT DISTINCT e FROM Endpoint e JOIN e.behaviorGroupActions bga JOIN bga.behaviorGroup.behaviors b " +
-                "WHERE e.enabled = TRUE AND b.eventType = :eventType AND bga.behaviorGroup.accountId = :accountId";
+                "WHERE e.enabled = TRUE AND b.eventType = :eventType AND (bga.behaviorGroup.accountId = :accountId OR bga.behaviorGroup.accountId IS NULL)";
 
         return sessionFactory.withStatelessSession(statelessSession -> {
             return statelessSession.createQuery(query, Endpoint.class)
                     .setParameter("eventType", eventType)
                     .setParameter("accountId", tenant)
                     .getResultList()
-                    .onItem().call(endpoints -> loadProperties(endpoints, true));
+                    .onItem().call(endpoints -> loadProperties(endpoints, true))
+                    .onItem().transformToMulti(endpoints -> Multi.createFrom().iterable(endpoints))
+                    .onItem().transformToUniAndConcatenate(endpoint -> {
+                        if (endpoint.getAccountId() == null) {
+                            if (endpoint.getType() == EndpointType.EMAIL_SUBSCRIPTION) {
+                                return this.getOrCreateEmailSubscriptionEndpoint(
+                                        tenant,
+                                        // We don't want to affect the original properties
```

Comment:
How could the properties be affected? This method is called during a Kafka messages processing.

Reply:
Passing the properties directly breaks the tests that uses this part. I will do some tests later to check what it does in the database

Reply:
This is the error it throws:
```
2022-01-05 10:12:54,878 INFO  [com.red.clo.not.eve.EventConsumer] (vert.x-eventloop-thread-23) Could not process the payload: {"bundle":"my-bundle","application":"policies-lifecycle-test","event_type":"all","timestamp":"2022-01-05T10:12:54.837302","account_id":"tenant","context":"{}","events":[{"metadata":{},"payload":"{}"}],"version":"v1.0.0","recipients":[]}: org.hibernate.LazyInitializationException: Unable to perform requested lazy initialization [com.redhat.cloud.notifications.models.EmailSubscriptionProperties.endpoint] - no session and settings disallow loading outside the Session
        at org.hibernate.bytecode.enhance.spi.interceptor.EnhancementHelper.throwLazyInitializationException(EnhancementHelper.java:275)
        at org.hibernate.bytecode.enhance.spi.interceptor.EnhancementHelper.performWork(EnhancementHelper.java:165)
        at org.hibernate.bytecode.enhance.spi.interceptor.LazyAttributeLoadingInterceptor.loadAttribute(LazyAttributeLoadingInterceptor.java:77)
        at org.hibernate.bytecode.enhance.spi.interceptor.LazyAttributeLoadingInterceptor.fetchAttribute(LazyAttributeLoadingInterceptor.java:73)
        at org.hibernate.bytecode.enhance.spi.interceptor.LazyAttributeLoadingInterceptor.handleRead(LazyAttributeLoadingInterceptor.java:54)
        at org.hibernate.bytecode.enhance.spi.interceptor.AbstractInterceptor.readObject(AbstractInterceptor.java:153)
        at com.redhat.cloud.notifications.models.EmailSubscriptionProperties.$$_hibernate_read_endpoint(EmailSubscriptionProperties.java)
        at com.redhat.cloud.notifications.models.EmailSubscriptionProperties.$$_hibernate_write_endpoint(EmailSubscriptionProperties.java)
        at com.redhat.cloud.notifications.models.EndpointProperties.setEndpoint(EndpointProperties.java:46)
        at com.redhat.cloud.notifications.db.EndpointResources.lambda$createEndpoint$0(EndpointResources.java:51)
```

Luckily it's noisy because we are in a stateless session.

Reply:
Wouldn't that work instead of calling `getOrCreateEmailSubscriptionEndpoint`?

```
                    .invoke(endpoint -> {
                        if (endpoint.getAccountId() == null) {
                            if (endpoint.getType() == EndpointType.EMAIL_SUBSCRIPTION) {
                                endpoint.setAccountId(tenant);
                            } else {
                                LOGGER.warnf("Invalid endpoint configured in default behavior group: %s", endpoint.getId());
                            }
                        }
                    })
```
Since the stateless session is used, calling `endpoint.setAccountId` should not affect the DB in any way.

Reply:
If that works, we could even avoid the `Multi` transformation.

Reply:
```
            return statelessSession.createQuery(query, Endpoint.class)
                    .setParameter("eventType", eventType)
                    .setParameter("accountId", tenant)
                    .getResultList()
                    .onItem().call(endpoints -> loadProperties(endpoints, true))
                    .invoke(endpoints -> {
                        for (Endpoint endpoint : endpoints) {
                            if (endpoint.getAccountId() == null) {
                                if (endpoint.getType() == EndpointType.EMAIL_SUBSCRIPTION) {
                                    endpoint.setAccountId(tenant);
                                } else {
                                    LOGGER.warnf("Invalid endpoint configured in default behavior group: %s", endpoint.getId());
                                }
                            }
                        }
                    });
```

Reply:
Yes, I think that's fine - We don't really need to create it in the database just yet.

Reply:
Pushed the changes